src/elements/content-sidebar/activity-feed/comment/BaseComment.js

```diff
@@ -157,23 +154,32 @@ const BaseComment = (props: BaseCommentProps) => {
         onSelect(false);
     };
 
-    const handleMessageUpdate = ({
-        id: messageID,
-        text,
-        hasMention,
-    }: {
-        hasMention: boolean,
-        id: string,
-        text: string,
-    }): void => {
-        onEdit(messageID, text, undefined, hasMention, permissions);
+    const handleMessageUpdate = ({ id: messageID, text, hasMention }) => {
+        // Since we have to pass onCommentEdit through annotations (to Replies), onAnnotationEdit essentially overrides onCommentEdit
+        if (onAnnotationEdit) {
+            onAnnotationEdit({ id: messageID, text, permissions });
+        } else if (onCommentEdit) {
+            onCommentEdit({
+                id: messageID,
+                text,
+                hasMention,
+                permissions,
+            });
+        }
         commentFormSubmitHandler();
     };
 
     const handleStatusUpdate = (selectedStatus: FeedItemStatus): void => {
-        onEdit(id, undefined, selectedStatus, false, permissions);
+        if (onAnnotationEdit) {
+            onAnnotationEdit({ id, permissions });
+        } else if (onCommentEdit) {
+            onCommentEdit({ id, status: selectedStatus, hasMention: false, permissions });
```

Comment:
We are passing an object here instead of parameters. This will cause the onCommentEdit method not to work properly. This ultimately calls this https://github.com/box/box-ui-elements/blob/master/src/api/Feed.js#L1907

Reply:
Also for the status update for annotations we will need to pass the actual status too right? Let's go back and verify that all the pieces are here for this to work.

Reply:
Oops, yep, I think you're right about the issue with onCommentEdit there. If we don't want to change it to an object all the way up the chain (although I'm tempted), I need to spread the args in ActivityFeed, like this:
```
onCommentEdit={hasCommentPermission ? props => onCommentUpdate(...props) : noop}
```

Reply:
About annotation status, `onAnnotationEdit` doesn't take `status`. We'd need to look into how annotations handle `status`---but I wasn't sure that was within the scope of this PR.